lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -218,15 +220,23 @@ public function fetch($fetchMode = null)
                 if ($result instanceof \stdClass) {
                     $result = $this->castObject($result, $className, $ctorArgs);
                 }
-
-                return $result;
+                break;
             case \PDO::FETCH_NUM:
-                return db2_fetch_array($this->_stmt);
+                $result = db2_fetch_array($this->_stmt);
+                break;
             case \PDO::FETCH_OBJ:
-                return db2_fetch_object($this->_stmt);
+                $result = db2_fetch_object($this->_stmt);
+                break;
             default:
                 throw new DB2Exception("Given Fetch-Style " . $fetchMode . " is not supported.");
         }
+
+        list($errorMessage, $errorCode) = $this->errorInfo();
+        if (false === $result && !empty($errorMessage)) {
+            throw new DB2Exception("Error during fetch: " . $errorMessage);
```

Comment:
I'm not quite sure about this. The interface does not allow such exceptions here. I suppose the correct way would be to check the result for `false` and then fetch the error information via `errorInfo()` manually. Otherwise it does not fulfill the contract.